src/api/index.js

```diff
@@ -264,30 +271,108 @@ export function rebuildPipelineRun(namespace, payload) {
 }
 
 export function getCredentials(namespace) {
-  const uri = getAPI('credentials', { namespace });
+  const queryParams = {
+    fieldSelector: 'type=kubernetes.io/basic-auth'
+  };
+  const uri = getKubeAPI('secrets', { namespace }, queryParams);
   return get(uri);
 }
 
 export function getCredential(id, namespace) {
-  const uri = getAPI('credentials', { name: id, namespace });
+  const uri = getKubeAPI('secrets', { name: id, namespace });
   return get(uri);
 }
 
 export function createCredential({ id, ...rest }, namespace) {
-  const uri = getAPI('credentials', { namespace });
+  const uri = getKubeAPI('secrets', { namespace });
   return post(uri, { id, ...rest });
 }
 
 export function updateCredential({ id, ...rest }, namespace) {
-  const uri = getAPI('credentials', { name: id, namespace });
+  const uri = getKubeAPI('secrets', { name: id, namespace });
   return put(uri, { id, ...rest });
 }
 
 export function deleteCredential(id, namespace) {
-  const uri = getAPI('credentials', { name: id, namespace });
+  const uri = getKubeAPI('secrets', { name: id, namespace });
   return deleteRequest(uri);
 }
 
+export function getServiceAccount({ name, namespace }) {
+  const uri = getKubeAPI('serviceaccounts', { name, namespace });
+  return get(uri);
+}
+
+export async function patchServiceAccount({
+  serviceAccountName,
+  namespace,
+  secretName
+}) {
+  const uri = getKubeAPI('serviceaccounts', {
+    name: serviceAccountName,
+    namespace
+  });
+  const patch1 = await patchAddSecret(uri, secretName);
+  return patch1;
+}
+
+// Get list of service accounts where secrets patched to, returns service account in full
```

Comment:
where secrets are* patched to

returns service accounts* as a list surely? not one service account in full

Reply:
Returns a list of service accounts

Reply:
My comment was about the comment 😸